core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -479,6 +479,29 @@ public void dayOfYear() {
     assertEquals(integerValue(220), eval(expression));
   }
 
+  @Test
+  public void day_Of_Year() {
+    when(nullRef.type()).thenReturn(DATE);
+    when(missingRef.type()).thenReturn(DATE);
+    assertEquals(nullValue(), eval(DSL.day_of_year(nullRef)));
+    assertEquals(missingValue(), eval(DSL.day_of_year(missingRef)));
+
+    FunctionExpression expression = DSL.day_of_year(DSL.literal(new ExprDateValue("2020-08-07")));
+    assertEquals(INTEGER, expression.type());
+    assertEquals("day_of_year(DATE '2020-08-07')", expression.toString());
+    assertEquals(integerValue(220), eval(expression));
+
+    expression = DSL.day_of_year(DSL.literal("2020-08-07"));
+    assertEquals(INTEGER, expression.type());
+    assertEquals("day_of_year(\"2020-08-07\")", expression.toString());
+    assertEquals(integerValue(220), eval(expression));
+
+    expression = DSL.day_of_year(DSL.literal("2020-08-07 01:02:03"));
```

Comment:
What if you give it an invalid string (random characters, just time etc.)

Reply:
Think edge cases: 29th or 30th day of February, the year 1200, the year 4000, 13th month, etc...

Reply:
for something fun: get a value of 366 (on a leap year) since that's the maximum value.

Reply:
Fixed in 09e5a3637655c6deb1f65a9282291c8cce0e4a4e and https://github.com/Bit-Quill/opensearch-project-sql/pull/173/commits/6c4d02c85ec5eccbb9eaed11952fc18d32eb29a2